BlazarUI/app/scripts/components/shared/ModuleModal.jsx

```diff
@@ -0,0 +1,83 @@
+import React, {Component, PropTypes} from 'react';
+import Immutable from 'immutable';
+import Modal from 'react-bootstrap/lib/Modal';
+import Button from 'react-bootstrap/lib/Button';
+import Checkbox from './Checkbox.jsx';
+import $ from 'jquery';
+
+class ModuleModal extends Component {
+
+  renderModule(moduleObject) {
+    let moduleId = moduleObject.get('id');
+    let moduleName = moduleObject.get('name');
+
+    return (
+      <div>
+        <Checkbox
+          label={moduleName}
+          name='module-checkbox'
+          value={moduleId}
+          checked={true} />
+        <br />
+      </div>
+    );
+  }
+
+  renderDownstreamToggle() {
+    return (
+      <Checkbox
+        label='Build Downstream Modules'
+        name='downstream-checkbox' />
+    );
+  }
+
+  renderModuleList() {
+    return (
+      <div className='module-modal--container'>
+        <div className='module-modal--modules'>
+          <span className='module-modal--title'>Pick Modules</span>
+          {this.props.modules.map(this.renderModule)}
+        </div>
+        <div className='module-modal--downstream'>
+          <span className='module-modal--title'>Toggle Option</span><br />
```

Comment:
not that `<br>` tags are wrong, I'd stick with using `<p>` tags if you want to create a block, and CSS for spacing tweaks, for the most part.